qualysvulnerabilitymanagement_connector.py

```diff
@@ -599,27 +600,34 @@ def _handle_scan_summary(self, param):
             self.debug_print("REST call fail while scan summary")
             return action_result.get_status()
 
-        response = response["SCAN_SUMMARY_OUTPUT"]["RESPONSE"]
+        scan_response: OrderedDict = response["SCAN_SUMMARY_OUTPUT"]["RESPONSE"]
 
         summary = action_result.update_summary({})
-        if response.get("SCAN_SUMMARY_LIST"):
-            response = response["SCAN_SUMMARY_LIST"]
-            if not isinstance(response.get("SCAN_SUMMARY"), list):
-                response["SCAN_SUMMARY"] = [response["SCAN_SUMMARY"]]
-            for scan in range(len(response["SCAN_SUMMARY"])):
-                if not isinstance(response["SCAN_SUMMARY"][scan].get("HOST_SUMMARY"), list):
-                    response["SCAN_SUMMARY"][scan]["HOST_SUMMARY"] = [response["SCAN_SUMMARY"][scan]["HOST_SUMMARY"]]
-
-                for host in range(len(response["SCAN_SUMMARY"][scan]["HOST_SUMMARY"])):
-
-                    res_data = response["SCAN_SUMMARY"][scan]["HOST_SUMMARY"][host]
-                    response["SCAN_SUMMARY"][scan]["HOST_SUMMARY"][host]["IP"] = res_data.pop("#text")
-                    response["SCAN_SUMMARY"][scan]["HOST_SUMMARY"][host]["CATEGORY"] = res_data.pop("@category")
-                    response["SCAN_SUMMARY"][scan]["HOST_SUMMARY"][host]["TRACKING_METHOD"] = res_data.pop("@tracking")
-
-            for data in response["SCAN_SUMMARY"]:
+        if "SCAN_SUMMARY_LIST" in scan_response:
+            scan_summary_payload: OrderedDict = scan_response["SCAN_SUMMARY_LIST"]
+
+            if not isinstance(scan_summary_payload.get("SCAN_SUMMARY"), list):
+                scan_summary_payload["SCAN_SUMMARY"] = [scan_summary_payload["SCAN_SUMMARY"]]
+
+            scan_summaries: list[OrderedDict] = scan_summary_payload["SCAN_SUMMARY"]
+            for scan_summary in scan_summaries:
+
+                # Not every scan summary contains host summary and these
+                # summaries should be skipped.
+                if "HOST_SUMMARY" not in scan_summary:
+                    continue
+
+                if not isinstance(scan_summary.get("HOST_SUMMARY"), list):
+                    scan_summary["HOST_SUMMARY"] = [scan_summary["HOST_SUMMARY"]]
```

Comment:
Does api docs allow for HOST_SUMMARY to be a list and something else in some cases? Or is it just when there is no such field?

Reply:
I believe it can be a single object if there is only a single host summary

Reply:
There is data on that in docs, https://qualysguard.qg2.apps.qualys.com/qwebhelp/fo_portal/api_doc/scans/vm_scans/scan_summary.htm Let's leave it as it is to keep it safe